### ISSUE TITLE
Incorrect missing NAT error on imported VPC

### ISSUE THREAD

**Issue by mdnorman** [sst/ion#895]
I'm importing a VPC using the following:
```
const vpc = sst.aws.Vpc.get("Vpc", vpcId);
```

Then using it in a cluster with a service:
```
const cluster = new sst.aws.Cluster("Cluster", { vpc });

cluster.addService("Service", {...})
```

However, it fails with the error:
```
The VPC configured for the service does not have NAT enabled. Enable NAT by configuring "nat" on the "sst.aws.Vpc" component.
```

I definitely have a NAT enabled on this VPC. It just wasn't created with SST ION.

**Comment by mdnorman** [comment]
Looks like the problem is here:
https://github.com/sst/ion/blob/8ddefa54a2d8915a176d3dea4a7f2dcfa4008536/platform/src/components/aws/vpc.ts#L544

this pattern may be created with SST, but any existing VPCs will likely have subjects named private in whatever case happened to be used at the time

**Comment by mdnorman** [comment]
[This code](https://github.com/sst/ion/blob/8ddefa54a2d8915a176d3dea4a7f2dcfa4008536/platform/src/components/aws/vpc.ts#L576) seems to attempt to find public subnets that connect to the NAT gateway. I would have expected the private subnets to connect to NAT gateway, not public. Am I missing something?

**Comment by mdnorman** [comment]
In addition, it seems to be expecting the names to be `*Private*` and `*Public*`, but an imported VPC may not have that convention.

**Comment by fwang** [comment]
> `The VPC configured for the service does not have NAT enabled. Enable NAT by configuring "nat" on the "sst.aws.Vpc" component.`
Fixed in v3.1.16

> it seems to be expecting the names to be *Private* and *Public*, but an imported VPC may not have that casing.
`sst.aws.Vpc.get()` is meant to be used to reference an `sst.aws.Vpc` created in another stage. If the VPC was manually created, you can directly pass it's id and subnets into other components.
